crates/iceberg/src/spec/values.rs

```diff
@@ -403,12 +403,26 @@ impl Datum {
                 }
             }
             PrimitiveType::Int => PrimitiveLiteral::Int(i32::from_le_bytes(bytes.try_into()?)),
-            PrimitiveType::Long => PrimitiveLiteral::Long(i64::from_le_bytes(bytes.try_into()?)),
+            PrimitiveType::Long => {
```

Comment:
Relevant section from the spec: https://iceberg.apache.org/spec/#schema-evolution

Reply:
I first did only the V2 transformations for now. I've added a test for the float and decimal case as well 👍